native/libcst/src/parser/grammar.rs

```diff
@@ -0,0 +1,2930 @@
+            = eq:lit("=") ex:expression() {
+                (make_assign_equal(eq), ex)
+            }
+
+        rule if_stmt() -> If<'a>
+            = i:lit("if") a:named_expression() col:lit(":") b:block() elif:elif_stmt() {
+                make_if(i, a, col, b, Some(OrElse::Elif(elif)), false)
+            }
+            / i:lit("if") a:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_if(i, a, col, b, el.map(OrElse::Else), false)
+            }
+
+        rule elif_stmt() -> If<'a>
+            = i:lit("elif") a:named_expression() col:lit(":") b:block() elif:elif_stmt() {
+                make_if(i, a, col, b, Some(OrElse::Elif(elif)), true)
+            }
+            / i:lit("elif") a:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_if(i, a, col, b, el.map(OrElse::Else), true)
+            }
+
+        rule else_block() -> Else<'a>
+            = el:lit("else") col:lit(":") b:block() {
+                make_else(el, col, b)
+            }
+
+        rule while_stmt() -> While<'a>
+            = kw:lit("while") test:named_expression() col:lit(":") b:block() el:else_block()? {
+                make_while(kw, test, col, b, el)
+            }
+
+        rule for_stmt() -> For<'a>
+            = f:lit("for") t:star_targets() i:lit("in") it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(None, f, t, i, it, c, b, el)
+            }
+            / asy:tok(Async, "ASYNC") f:lit("for") t:star_targets() i:lit("in")
+                it:star_expressions()
+                c:lit(":") b:block() el:else_block()? {
+                    make_for(Some(asy), f, t, i, it, c, b, el)
+            }
+
+        rule with_stmt() -> With<'a>
+            = kw:lit("with") items:separated(<with_item()>, <comma()>)
+                col:lit(":") b:block() {
+                    make_with(None, kw, comma_separate(items.0, items.1, None), col, b)
+            }
+            / asy:tok(Async, "ASYNC") kw:lit("with") items:separated(<with_item()>, <comma()>)
+                col:lit(":") b:block() {
+                    make_with(Some(asy), kw, comma_separate(items.0, items.1, None), col, b)
+            }
+
+        rule with_item() -> WithItem<'a>
+            = e:expression() a:lit("as") t:star_target() &(lit(",") / lit(":")) {
+                make_with_item(e, Some(a), Some(t))
+            }
+            / e:expression() {
+                make_with_item(e, None, None)
+            }
+
+        rule named_expression() -> Expression<'a>
+            = a:name() op:lit(":=") b:expression() {
+                Expression::NamedExpr(make_named_expr(a, op, b))
+            }
+            / e:expression() !lit(":=") { e }
+
+        rule annotated_rhs() -> Expression<'a>
+            = yield_expr() / star_expressions()
+
+        rule global_stmt() -> Global<'a>
+            = kw:lit("global") init:(n:name() c:comma() {(n, c)})* last:name() {
```

Comment:
[nit] it sems like in most cases where the official grammar has `','.something+`, we've been able to use a similar form involving comma-separated stuff. Does that not apply here?

I think what we have is equivalent but the grammar diverges

Reply:
The official grammar has the luxury of discarding all those separators. We need to refer to them later, which is why this is a bit more complex. Eventually I cooked up the `separated` and `separated_trailer` helper meta-rules, but
1. I'm not sure they improve readability
2. I was too lazy to go back and refactor every occurrence of this pattern. This is tech debt TODO